src/app/services/upload/upload-result.interface.ts

```diff
@@ -1,10 +1,13 @@
 export interface UploadResult {
-  id: number;
+  id: string;
+  file: File;
   orginalName: string;
   cleanName: string;
   name: string;
   extension: string;
-  size_in_mb: number;
-  publicUrl: string;
-  filePath: string;
+  sizeInMb: number;
+  publicUrl?: string;
+  filePath?: string;
+  uploadProgressInPercent: number;
+  uploadStatus: number | 'canceled';
```

Comment:
i would not specify the type like this - what does `number` mean? not very readable to me.